go/capture/driver/airbyte/driver.go

```diff
@@ -443,29 +295,18 @@ func (d driver) Pull(stream pc.Driver_PullServer) error {
 				}
 			}
 		},
-		// Expect to decode Airbyte messages.
-		connector.NewJSONOutput(
-			func() interface{} { return new(airbyte.Message) },
-			func(i interface{}) error {
-				if rec := i.(*airbyte.Message); rec.Log != nil {
-					logger.Log(airbyteToLogrusLevel(rec.Log.Level), nil, rec.Log.Message)
-				} else if rec.State != nil {
-					return pc.WritePullCheckpoint(stream, &resp,
-						&pf.DriverCheckpoint{
-							DriverCheckpointJson: rec.State.Data,
-							Rfc7396MergePatch:    rec.State.Merge,
-						})
-				} else if rec.Record != nil {
-					if b, ok := streamToBinding[rec.Record.Stream]; ok {
-						return pc.StagePullDocuments(stream, &resp, b, rec.Record.Data)
-					}
-					return fmt.Errorf("connector record with unknown stream %q", rec.Record.Stream)
+		connector.NewProtoOutput(
+			func() proto.Message { return new(pc.PullResponse) },
+			func(m proto.Message) error {
+				var pullResp = m.(*pc.PullResponse)
+				if c := pullResp.Checkpoint; c != nil {
+					return pc.WritePullCheckpoint(stream, &resp, c)
+				} else if d := pullResp.Documents; d != nil {
```

Comment:
The translation from airbyte message to proto message is done in the Rust connector. However, staging / combining the messages are done in Go side. It this OK? Or should the boundary between the Rust/Go be shifted?